### PR TITLE
chore: bump to astro v2

### LINKED ISSUE #40: Migrate to Astro v2
Migration guide: https://docs.astro.build/en/guides/upgrade-to/v2/

Current Astro version on this project is 1.9.2:

https://github.com/teknologi-umum/tgif/blob/b270d7bd11b49464574ee43c2104f8d1b26adeea/package.json#L22

### PR DESCRIPTION
**Pull request by elianiva**
closes #40

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,43 +13,43 @@
...
-		"prettier-plugin-astro": "^0.7.2",
+		"@typescript-eslint/eslint-plugin": "^5.61.0",
+		"@typescript-eslint/parser": "^5.61.0",
+		"@vitest/coverage-c8": "^0.33.0",
+		"autoprefixer": "^10.4.14",
+		"cssnano": "^6.0.1",
+		"eslint": "^8.44.0",
+		"postcss": "^8.4.25",
+		"prettier": "^3.0.0",
+		"prettier-plugin-astro": "^0.10.0",
```
Comment (aldy505) [inline package.json]: Regarding this, shouldn't we use `@teknologi-umum/eslint-config-typescript`?

**Comment by aldy505** [comment]
The prettier plugin's broken on CI

**Comment by aldy505** [comment]
@elianiva Mind if I continue this PR by pushing directly to it?

**Comment by elianiva** [comment]
lanjutin aja

**Comment by aldy505** [comment]
@elianiva an error during build. I don't know why

```
 error   ENOENT: no such file or directory, scandir '***/tgif/dist/chunks/public/chats'
Error: ENOENT: no such file or directory, scandir '***/tgif/dist/chunks/public/chats'
 ELIFECYCLE  Command failed with exit code 1.
```

**Pull request merged**